src/Orleans.Streaming/QueueBalancer/LeaseBasedQueueBalancer.cs

```diff
@@ -1,453 +1,497 @@
+        _queueSelector = new RoundRobinSelector<QueueId>(allQueues);
+        await base.Initialize(queueMapper);
+        StartMaintenanceTasks();
+
+        void StartMaintenanceTasks()
         {
-            var options = services.GetOptionsByName<LeaseBasedQueueBalancerOptions>(name);
-            var leaseProvider = services.GetKeyedService<ILeaseProvider>(name)
-                ?? services.GetRequiredService<ILeaseProvider>();
-            return ActivatorUtilities.CreateInstance<LeaseBasedQueueBalancer>(services, name, options, leaseProvider);
+            using var _ = new ExecutionContextSuppressor();
+            _leaseAcquisitionTimerTask = PeriodicallyAcquireLeasesToMeetResponsibility();
+            _leaseMaintenanceTimerTask = PeriodicallyMaintainLeases();
         }
+    }
 
-        /// <inheritdoc/>
-        public override Task Initialize(IStreamQueueMapper queueMapper)
-        {
-            if (base.Cancellation.IsCancellationRequested) throw new InvalidOperationException("Cannot initialize a terminated balancer.");
-            if (queueMapper == null)
-            {
-                throw new ArgumentNullException(nameof(queueMapper));
-            }
-            var allQueues = queueMapper.GetAllQueues().ToList();
-            this.allQueuesCount = allQueues.Count;
+    /// <inheritdoc/>
+    public override async Task Shutdown()
+    {
+        if (Cancellation.IsCancellationRequested) return;
 
-            //Selector default to round robin selector now, but we can make a further change to make selector configurable if needed.  Selector algorithm could 
-            //be affecting queue balancing stablization time in cluster initializing and auto-scaling
-            this.queueSelector = new RoundRobinSelector<QueueId>(allQueues);
-            return base.Initialize(queueMapper);
-        }
+        // Stop acquiring and renewing leases.
+        _leaseMaintenanceTimer.Dispose();
+        _leaseAcquisitionTimer.Dispose();
+        await Task.WhenAll(_leaseMaintenanceTimerTask, _leaseAcquisitionTimerTask);
 
-        /// <inheritdoc/>
-        public override async Task Shutdown()
+        // Release all owned leases.
+        var shutdownTask = _executor.AddNext(async () =>
         {
-            if (base.Cancellation.IsCancellationRequested) return;
-            this.myQueues.Clear();
-            this.responsibility = 0;
-            this.leaseMaintenanceTimer?.Dispose();
-            this.leaseMaintenanceTimer = null;
-            this.leaseAquisitionTimer?.Dispose();
-            this.leaseAquisitionTimer = null;
-            await base.Shutdown();
-            //release all owned leases
-            await this.executor.AddNext(this.ReleaseLeasesToMeetResponsibility);
-        }
+            _responsibility = 0;
+            await ReleaseLeasesToMeetResponsibility();
+        });
 
-        /// <inheritdoc/>
-        public override IEnumerable<QueueId> GetMyQueues()
+        // Signal shutdown.
```

Comment:
During shutdown, we should:
1. Stop the timers (cancelling them so the tasks will complete).
2. Wait for the timer tasks to complete.
3. Release all leases